dicom_utils/types.py

```diff
@@ -12,108 +13,138 @@
-    r"""Produces an ImageType value from a dictionary of processed image type
-    fields (from :func:`process_image_type`).
+@dataclass(frozen=True)
+class ImageType:
+    """Container for DICOM metadata fields related to the image type.
+
+    Contains the following attributes:
 
-    ImageType is determined as follows:
-        1. ImageType.TOMO if ``NumberOfFrames > 1``
-        2. ImageType.SVIEW if ``pixels == "derived`` and (``"generated" in extras or flavor is not None``)
-        3. ImageType.2D otherwise
+        * ``"pixels"`` - First element of the IMAGE_TYPE field.
+        * ``"exam"`` - Second element of the IMAGE_TYPE field.
+        * ``"flavor"`` - Third element of the IMAGE_TYPE field.
+        * ``"extras"`` - Additional IMAGE_TYPE fields if available
+        * ``"NumberOfFrames"`` - Frame count (for TOMO images only)
+        * ``"model"`` - Manufacturer's model name
     """
-    # if fields 1 and 2 were missing, we know nothing
-    if not image_type:
-        return ImageType.UNKNOWN
-
-    pixels = image_type.get("pixels", "").lower() or ""
-    flavor = image_type.get("flavor", "").lower() or ""
-    exam = image_type.get("exam", "").lower() or ""
-    extras = image_type.get("extras", []) or []
-    num_frames = int(image_type.get("NumberOfFrames", 1) or 1)
-    machine = image_type.get("model", "").lower() or ""
-
-    # very solid rules
-    if num_frames > 1:
-        return ImageType.TOMO
-    if "original" in pixels:
-        return ImageType.NORMAL
-
-    # ok rules
-    if extras is not None and any("generated_2d" in x.lower() for x in extras):
-        return ImageType.SVIEW
-
-    # not good rules
-    if pixels == "derived" and exam == "primary" and machine == "fdr-3000aws" and flavor != "post_contrast":
-        return ImageType.SVIEW
-
-    return ImageType.NORMAL
+
+    pixels: str
+    exam: str
+    flavor: Optional[str] = None
+    extras: Optional[List[str]] = None
+    NumberOfFrames: Optional[int] = None
+    model: Optional[str] = None
+
+    def __bool__(self) -> bool:
+        return bool(self.pixels) and bool(self.exam)
+
+    def __post_init__(self):
+        if (nf := self.NumberOfFrames) is not None and not isinstance(nf, int):
+            object.__setattr__(self, "NumberOfFrames", int(nf))
+        assert self.NumberOfFrames is None or self.NumberOfFrames > 0
+
+    def simple_repr(self) -> str:
+        s = "|".join(x for x in (self.pixels, self.exam))
+        if self.flavor is not None:
+            s += f"|{self.flavor}" if self.flavor else "|''"
+        for elem in self.extras or []:
+            if not elem or elem.isdigit():
+                continue
+            s += f"|{elem}"
+        return s
+
+    @classmethod
+    def from_dicom(cls, dcm: Dicom) -> "ImageType":
+        result: Dict[str, Any] = {}
+        result["NumberOfFrames"] = dcm.get("NumberOfFrames", 1)
+        if (model := dcm.get(0x00081090, None)) is not None:
```

Comment:
I think for readability, lines 86 through 89 could be replaced with: 
`result["model"] = ds.get("ManufacturerModelName", "")`